tick/inference/linear_regression.py

```diff
@@ -0,0 +1,145 @@
+
+    record_every : `int`, default=1
+        Record history information when ``n_iter`` (iteration number) is
+        a multiple of ``record_every``
+
+    elastic_net_ratio : `float`, default=0.95
+        Ratio of elastic net mixing parameter with 0 <= ratio <= 1.
+        For ratio = 0 this is ridge (L2 squared) regularization
+        For ratio = 1 this is lasso (L1) regularization
+        For 0 < ratio < 1, the regularization is a linear combination
+        of L1 and L2.
+        Used in 'elasticnet' penalty
+
+    random_state : `int` seed, `RandomState` instance, or `None` (default)
+        The seed that will be used by stochastic solvers. Used in 'sgd',
+        'svrg', and 'sdca' solvers
+
+    Attributes
+    ----------
+    weights : `np.array`, shape=(n_features,)
+        The learned weights of the model (not including the intercept)
+
+    intercept : `float` or `None`
+        The intercept, if ``fit_intercept=True``, otherwise `None`
+    """
+
+    _solvers = {
+        'gd': GD,
+        'agd': AGD,
+        'svrg': SVRG
+    }
+
+    _attrinfos = {
+        "_actual_kwargs": {"writable": False}
+    }
+
+    @actual_kwargs
+    def __init__(self, fit_intercept=True, penalty='l2', C=1e3,
+                 solver='svrg', step=None, tol=1e-5, max_iter=100,
+                 verbose=False, warm_start=False,
+                 print_every=10, record_every=1,
+                 elastic_net_ratio=0.95, random_state=None):
+
+        self._actual_kwargs = LinearRegression.__init__.actual_kwargs
+        LearnerGLM.__init__(self, fit_intercept=fit_intercept,
+                            penalty=penalty, C=C,
+                            solver=solver, step=step, tol=tol,
+                            max_iter=max_iter, verbose=verbose,
+                            warm_start=warm_start, print_every=print_every,
+                            record_every=record_every,
+                            elastic_net_ratio=elastic_net_ratio,
+                            random_state=random_state)
+
+    def _construct_model_obj(self, fit_intercept=True):
+        return ModelLinReg(fit_intercept)
+
+    def predict(self, X):
+        """Predict class for given samples
+
+        Parameters
+        ----------
+        X : `np.ndarray` or `scipy.sparse.csr_matrix`, shape=(n_samples, n_features)
+            Features matrix to predict for.
+
+        Returns
+        -------
+        output : `np.array`, shape=(n_samples,)
+            Returns predicted values.
+        """
+        if not self._fitted:
+            raise ValueError("You must call ``fit`` before")
+        else:
+            X = self._safe_array(X)
+            z = X.dot(self.weights)
+            if self.intercept:
+                z += self.intercept
+            return z
+
+    def score(self, X, y):
+        """Returns the coefficient of determination R^2 of the fitted linear
+        regression model, computed on the given features matrix and labels.
```

Comment:
I would add the scikit learn description

>Best possible score is 1.0 and it can be negative (because the model can be arbitrarily worse). A constant model that always predicts the expected value of y, disregarding the input features, would get a R^2 score of 0.0.

For people that don't know much about this